src/ProxyClient/Varnish.php

```diff
@@ -125,28 +125,36 @@ public function refresh($url, array $headers = array())
 
         return $this;
     }
-
-    /**
-     * {@inheritdoc}
-     *
-     * @throws MissingHostException If a relative path is queued for purge/
-     *                              refresh and no base URL is set
-     *
-     */
-    protected function createRequest($method, $url, array $headers = array())
+    
+    protected function queueRequest($method, $url, array $headers = array())
     {
-        $request = parent::createRequest($method, $url, $headers);
-
-        // For purge and refresh, add a host header to the request if it hasn't
-        // been set
-        if (self::HTTP_METHOD_BAN !== $method
-            && '' == $request->getHeader('Host')
-        ) {
-            throw MissingHostException::missingHost($url);
+        if (self::HTTP_METHOD_BAN !== $method) {
+            // AND NO HOST IS SET
```

Comment:
This is somewhat problematic: I need to know here whether the request has a Host header. That header is determined from either the base URL or the endpoint of the request itself (in case of an absolute URL). I moved base URL handling into a `RequestQueue`, so I cannot check that here. I guess this logic needs to be moved back to the Varnish and Nginx client classes.